zk/stages/stage_sequence_execute_transactions.go

```diff
@@ -187,6 +195,10 @@ func attemptAddTransaction(
 	)
 
 	if err != nil {
+		if errors.Is(err, core.ErrGasLimitReached) {
+			log.Debug("Transaction gas limit reached", "txHash", transaction.Hash())
+			return nil, nil, txCounters, overflowGas, nil
```

Comment:
Although this looks correct, I believe we don't handle gas overflow returned value properly in the caller. At least that is what I experienced a while ago when tried to simulate gas overflow in this function. Can't remember though what exactly the issue was.
But if you tested it, lets merge it!